File: omnihealthmobileflutter/lib/domain/abstracts/auth_repository.dart
import '../entities/user_entity.dart';

abstract class AuthRepository {
  /// Login with email and password
  Future<UserEntity> login(LoginEntity loginEntity);

  /// Logout current user
  Future<void> logout();

  /// Get current user
  UserEntity? getCurrentUser();

  /// Send password reset email
  Future<void> sendPasswordResetEmail(String email);
}

File: adminpage/src/domain/repositories/auth.repository.ts
import type { User } from "../../shared/types";

export interface IAuthRepository {
  login(
    email: string,
    password: string
  ): Promise<{ user: User; accessToken: string; refreshToken: string }>;
  register(
    userData: any
  ): Promise<{ user: User; accessToken: string; refreshToken: string }>;
  getCurrentUser(): Promise<User>;
  refreshToken(): Promise<{ accessToken: string; refreshToken: string }>;
  logout(): Promise<void>;
}
